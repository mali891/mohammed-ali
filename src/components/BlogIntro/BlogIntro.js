import React from 'react'
import PropTypes from 'prop-types'

import { formatDate } from '../../functions'
import { styles } from './BlogIntro.style'

const BlogIntro = ({ children, className, date, length, excerpt, ...other }) => {
  return (
    <section css={styles} className="c-blog-intro" {...other}>
      <div className="c-blog-intro__metadata">
        <span className="c-blog-intro__metadata__content">Published on {formatDate(date)}</span>{' '}
        <span className="c-blog-intro__metadata__content">{length} read</span>
      </div>

      <div className="c-blog-intro__excerpt">
        <p>
          <em>{excerpt}</em>
        </p>
      </div>
    </section>
  )
}

BlogIntro.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  className: PropTypes.string,
  date: PropTypes.string.isRequired,
  excerpt: PropTypes.string.isRequired,
  length: PropTypes.string.isRequired
}

export default BlogIntro
