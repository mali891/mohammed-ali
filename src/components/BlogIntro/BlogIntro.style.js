import { css } from '@emotion/react'
import { SPACING, MEDIA_QUERIES, VIEWPORT_SIZES, FONT_SIZES, COLOURS } from '../../styles/settings'

export const styles = css`
  position: relative;

  &:after {
    content: '';
    height: 0.2rem;
    width: 100%;
    background-color: ${COLOURS.LIGHT};
    position: absolute;
    transform: translate(-50%, -50%);
    left: 50%;

    ${VIEWPORT_SIZES.map(
      viewport =>
        `@media (min-width: ${MEDIA_QUERIES[`${viewport}`]}) {
          bottom: calc(${SPACING[`${viewport}`].XL} / -2);
        }`
    )}
  }

  .c-blog-intro {
    &__metadata {
      width: 100%;
      display: flex;
      flex-direction: column;

      ${VIEWPORT_SIZES.map(
        viewport =>
          `@media (min-width: ${MEDIA_QUERIES[`${viewport}`]}) {
            margin-bottom: calc(${SPACING[`${viewport}`].XL} / 2);
          }`
      )}

      &__content {
        display: block;
        width: 100%;
        text-align: center;
        font-size: ${FONT_SIZES.XS};

        &:first-of-type {
          margin-bottom: 2rem;
        }
      }
    }
  }

  @media (min-width: ${MEDIA_QUERIES.TABLET}) {
    .c-blog-intro {
      &__metadata {
        flex-direction: row;
        justify-content: space-between;

        &__content {
          display: unset;
          width: unset;
        }
      }
    }
  }
`
