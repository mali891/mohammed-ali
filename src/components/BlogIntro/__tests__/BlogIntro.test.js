import React from 'react'
import { render } from '@testing-library/react'

import BlogIntro from '../BlogIntro'

const defaultProps = {
  date: '2030-01-01',
  length: '1 min',
  excerpt: 'lorem ipsum sit dolor amet'
}

describe('BlogIntro', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<BlogIntro {...defaultProps} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
