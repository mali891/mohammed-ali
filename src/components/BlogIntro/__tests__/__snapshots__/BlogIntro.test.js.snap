// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogIntro renders correctly 1`] = `
<DocumentFragment>
  <section
    class="c-blog-intro"
    css="[object Object]"
  >
    <div
      class="c-blog-intro__metadata"
    >
      <span
        class="c-blog-intro__metadata__content"
      >
        Published on 1st Jan 2030
      </span>
       
      <span
        class="c-blog-intro__metadata__content"
      >
        1 min read
      </span>
    </div>
    <div
      class="c-blog-intro__excerpt"
    >
      <p>
        <em>
          lorem ipsum sit dolor amet
        </em>
      </p>
    </div>
  </section>
</DocumentFragment>
`;
